src/openrct2/interface/InteractiveConsole.cpp

```diff
@@ -1924,6 +1924,110 @@ static int32_t ConsoleCommandProfilerStop(
     return 0;
 }
 
+static int32_t ConsoleCommandGoToEntity([[maybe_unused]] InteractiveConsole& console, [[maybe_unused]] const arguments_t& argv)
+{
+    if (argv.size() < 1)
+    {
+        console.WriteLineWarning("Too few arguments");
+        static_assert(EnumValue(EntityType::Count) == 13, "EntityType::Count changed, update console command!");
+        console.WriteLine("get_entity <entity ID> [quiet = suppress camera]");
+        return 1;
+    }
+    auto entityId = atoi(argv[0].c_str());
+    if (entityId > MAX_ENTITIES || entityId < 0)
+    {
+        console.WriteFormatLine("Entity number out of bounds. Maximum: %d", MAX_ENTITIES);
+        return 1;
+    }
+
+    bool quiet = false;
+    if (argv.size() >= 2)
+    {
+        quiet = true;
+    }
+
+    auto* entity = TryGetEntity(EntityId::FromUnderlying(entityId));
+
+    if (entity == nullptr || entity->Type == EntityType::Null)
+    {
+        console.WriteFormatLine("Could not find entity with ID %d", entityId);
+        return 1;
+    }
+
+    CoordsXYZ entityLocation = entity->GetLocation();
+    CoordsXYZ majorGrid = { entityLocation.x / 32, entityLocation.y / 32, entityLocation.z / 8 };
+    CoordsXYZ minorGrid = { entityLocation.x - majorGrid.x * 32, entityLocation.y - majorGrid.y * 32,
+                            entityLocation.z - majorGrid.z * 8 };
+
+    char vehicleExtraInfo[64] = "";
+    if (entity->Type == EntityType::Vehicle)
+    {
+        auto ride = entity->As<Vehicle>()->GetRide();
+        if (ride != nullptr)
```

Comment:
You can just use the GetHead function to get the train head. Will make this code simpler.

Reply:
Iterating over the trains is necessary to get the train number since Vehicle does not store that information. I can use GetHead to avoid iterating over trains that aren't the vehicle's though.